core/modules/map.rs

```diff
@@ -83,6 +83,8 @@ pub(crate) struct ModuleMap {
   pub(crate) loader: RefCell<Rc<dyn ModuleLoader>>,
   pub(crate) import_meta_resolve_cb: ImportMetaResolveCallback,
 
+  // TODO(mmastrac): temporarily crate-public to avoid RefCell on ContextState
+  pub(crate) exception_state: Rc<ExceptionState>,
```

Comment:
I might be nitpicking here, but seems trange to store `exception_state` on `ModuleMap`. Is it duplicated in `JsRuntimeState`?

Reply:
This is some leftover cruft that hasn't been fully refactored yet...

It's on the ContextState and ModuleMap for now, but ContextState still has a RefCell which makes it more expensive to access in the event loop. The ModuleMap technically only needs this because we're still threading `reportError` errors through it. 

Once I remove that path (especially if we can get to a single unhandled rejection reporting channel), we might be able to get rid of this entirely.

Reply:
Looked at this briefly and I think it's better to do in a follow-up pass. Even if I can remove it, the RefCell on ContextState will be a problem.